aeron-agent/src/main/java/io/aeron/agent/ArchiveEventEncoder.java

```diff
@@ -29,6 +29,198 @@ private ArchiveEventEncoder()
     {
     }
 
+    static <E extends Enum<E>> int encodeReplaySessionStateChange(
+        final UnsafeBuffer encodingBuffer,
+        final int offset,
+        final int captureLength,
+        final int length,
+        final E from,
+        final E to,
+        final long id,
+        final long recordingId,
+        final long position,
+        final String reason)
+    {
+        int encodedLength = encodeLogHeader(encodingBuffer, offset, captureLength, length);
+
+        encodingBuffer.putLong(offset + encodedLength, id, LITTLE_ENDIAN);
+        encodedLength += SIZE_OF_LONG;
+        encodingBuffer.putLong(offset + encodedLength, recordingId, LITTLE_ENDIAN);
+        encodedLength += SIZE_OF_LONG;
+        encodingBuffer.putLong(offset + encodedLength, position, LITTLE_ENDIAN);
+        encodedLength += SIZE_OF_LONG;
+
+        encodedLength += encodingBuffer.putStringAscii(offset + encodedLength, reason);
+
+        return encodeTrailingStateChange(encodingBuffer, offset, encodedLength, captureLength, from, to);
+    }
+
+    static <E extends Enum<E>> int replaySessionStateChangeLength(final E from, final E to, final String reason)
+    {
+        return stateTransitionStringLength(from, to) + (3 * SIZE_OF_LONG) + (SIZE_OF_INT + reason.length());
+    }
+
+    static <E extends Enum<E>> int encodeReplaySessionStarted(
+        final UnsafeBuffer encodingBuffer,
+        final int offset,
+        final int captureLength,
+        final int length,
+        final long sessionId,
+        final long controlSessionId,
+        final long correlationId,
+        final long streamId,
+        final long recordingId,
+        final long startPosition,
+        final String publicationChannel)
+    {
+        int encodedLength = encodeLogHeader(encodingBuffer, offset, captureLength, length);
+
+        encodingBuffer.putLong(offset + encodedLength, sessionId, LITTLE_ENDIAN);
+        encodedLength += SIZE_OF_LONG;
+        encodingBuffer.putLong(offset + encodedLength, controlSessionId, LITTLE_ENDIAN);
+        encodedLength += SIZE_OF_LONG;
+        encodingBuffer.putLong(offset + encodedLength, correlationId, LITTLE_ENDIAN);
+        encodedLength += SIZE_OF_LONG;
+        encodingBuffer.putLong(offset + encodedLength, streamId, LITTLE_ENDIAN);
+        encodedLength += SIZE_OF_LONG;
+        encodingBuffer.putLong(offset + encodedLength, recordingId, LITTLE_ENDIAN);
+        encodedLength += SIZE_OF_LONG;
+        encodingBuffer.putLong(offset + encodedLength, startPosition, LITTLE_ENDIAN);
+        encodedLength += SIZE_OF_LONG;
+
+        encodedLength += encodingBuffer.putStringAscii(offset + encodedLength, publicationChannel);
```

Comment:
This should be using something like `CommonEncoder.encodeTrailingString` to truncate the message if required.